### PR TITLE
Install nodelet plugin descriptor file as well.

### LINKED ISSUE #55: Skipping XML documentation
Hello 

I am using this package with my project.
I met error bellows 

`Skipping XML Document “/opt/ros/indigo/share/gmapping/nodelet_plugins.xml” which had no Root Element. This likely means the XML is malformed or missing.`

After I had manually copied 'nodelet_plugins.xml' file from here, then this error was disappeared.

I don't know root cause exactly, but I guess something is missed on CMakeLists.xml.

Thanks.

### PR DESCRIPTION
**Pull request by gavanderhoorn**
This seems to have been missed in #41.

Related ROS Answers question: [pointcloud_to_laserscan: "Skipping XML Document..."](https://answers.ros.org/question/274849) (note that the problem is here, in `gmapping`, not in `pointcloud_to_laserscan`).

Fixes #55.

Without this, from-source builds in `devel` spaces will work fine, but the binary debians will not contain the required plugin descriptor file.

This will need a new release on all platforms.

### CODE REVIEW

**Review by wjwwood — APPROVED** [review]
lgtm, but I'll leave to @vrabaud to merge and release.

**Comment by gavanderhoorn** [comment]
@vrabaud: very gentle 🛎.

**Comment by furushchev** [comment]
+1 to be merged.
On kinetic, just installing this package collapses all nodelet functions.

**Comment by josiahls** [comment]
Hi, I am having this same issue:
`[ERROR] [1510694438.414259937]: Skipping XML Document "/opt/ros/kinetic/share/gmapping/nodelet_plugins.xml" which had no Root Element.  This likely means the XML is malformed or missing.`

This occurs when trying do an rviz turtlebot minimal launch. Won't let rvis continue without this. I have the gmapping version 1.3.9, and I looked at the file location and there is still no nodelet_plugins.xml. Is this fix in version 1.3.9 or is this still waiting for a newer release?

**Comment by gavanderhoorn** [comment]
@josiahls wrote:
> I have the gmapping version 1.3.9, and I looked at the file location and there is still no nodelet_plugins.xm

this PR hasn't been merged yet, so it is not part of any release.

Building it from source in an overlay workspace would allow you to use the fixed version immediately.

**Comment by mikaelarguedas** [comment]
:+1: it would be great to get this merged and released

**Comment by garyservin** [comment]
+1

**Comment by fmessmer** [comment]
:+1:

**Comment by gavanderhoorn** [comment]
Could this get merged? I'm not really personally affected by this, but there are a nr of questions popping up on ROS Answers about issues that in the end turn out to be caused by `pluginlib` giving up because of the missing xml file.

ros/pluginlib#85 mitigates that, but only so much.

**Comment by mikaelarguedas** [comment]
[pluginlib's fix](ros/pluginlib#85) has been released and synced in lunar. It should make it to the main repo for kinetic in the [upcoming sync](https://discourse.ros.org/t/preparing-for-kinetic-sync-2017-11-29/3275).

But I agree with @gavanderhoorn it would be really great to get this merged ans released soon :+1:

**Comment by elhayra** [comment]
+1

**Comment by eraserpencil** [comment]
Pardon my ignorance. Has the fix been synced? How do I check if it has?

**Comment by gavanderhoorn** [comment]
@eraserpencil: no, it hasn't.

> How do I check if it has?

at the very least this PR should have a status `merged`, instead of `open`.

**Comment by gavanderhoorn** [comment]
@wjwwood: would it be an option to merge this in? With ros/pluginlib#85 merged it's less of an issue in case you don't want to use `gmapping`, but without manually adding the changes in this PR, the `gmapping` nodelet of even a source build is unusable.

**Comment by QinZiwen** [comment]
I am having this same issue:
[ERROR] [1516671310.130156638]: Skipping XML Document "/opt/ros/indigo/share/gmapping/nodelet_plugins.xml" which had no Root Element.  This likely means the XML is malformed or missing.

I did not find a solution to this problem, and I use indigo

**Comment by wjwwood** [comment]
@gavanderhoorn it is an option, but I'd rather the maintainer handle this (I have too many packages to do this kind of thing on all of them):

> lgtm, but I'll leave to @vrabaud to merge and release.

**Comment by gavanderhoorn** [comment]
I agree completely, but it seems @vrabaud is a bit busy :)

**Comment by skyinzero** [comment]
@gavanderhoorn excuse me,where is gmapping/CMakeLists.txt. I mean, the path of gmapping/CMakeLists.txt. thx!

**Comment by gavanderhoorn** [comment]
[here](https://github.com/ros-perception/slam_gmapping/tree/hydro-devel/gmapping)?

But know that you won't find it if you installed `gmapping` using `apt-get`.

**Comment by skyinzero** [comment]
@gavanderhoorn got it. thx!!!! : )

**Comment by mikaelarguedas** [comment]
> wjwwood: would it be an option to merge this in? With ros/pluginlib#85 merged it's less of an issue in case you don't want to use gmapping, but without manually adding the changes in this PR, the gmapping nodelet of even a source build is unusable.

I got green light from @vrabaud to merge and release this, as 1.3.9 is [in all active distros](http://repositories.ros.org/status_page/compare_indigo_kinetic_lunar.html?q=gmapping) I'll release on all of them so that it makes it into the next sync

**Comment by mikaelarguedas** [comment]
Released in [Indigo](https://github.com/ros/rosdistro/pull/16817), [Kinetic](https://github.com/ros/rosdistro/pull/16818) and [Lunar](https://github.com/ros/rosdistro/pull/16819)

**Comment by QinZiwen** [comment]
@mikaelarguedas You mean is that this problem can be solved by reinstalling ros-indigo?
Problem is:
```
[ERROR]: Skipping XML Document "/opt/ros/indigo/share/gmapping/nodelet_plugins.xml" which had no Root Element. This likely means the XML is malformed or missing.
```
Reinstall ros-indigo:
```
sudo apt-get install ros-indigo-desktop
```

**Comment by gavanderhoorn** [comment]
@QinZiwen: no, you just need to install the upgraded packages that will be made available in the next public sync, or use the [shadow repository](http://wiki.ros.org/ShadowRepository) if you can't wait.

**Comment by gavanderhoorn** [comment]
Thanks @mikaelarguedas.

**Comment by QinZiwen** [comment]
Thanks @gavanderhoorn

**Pull request merged**
